admin/server/reports.go

```diff
@@ -57,48 +57,39 @@ func (s *Server) GetReportMeta(ctx context.Context, req *adminv1.GetReportMetaRe
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	if len(req.EmailRecipients) == 0 {
-		return &adminv1.GetReportMetaResponse{
-			BaseUrls: &adminv1.GetReportMetaResponse_URLs{
-				OpenUrl:   s.admin.URLs.WithCustomDomain(org.CustomDomain).ReportOpen(org.Name, proj.Name, req.Report, req.ExecutionTime.AsTime()),
-				ExportUrl: s.admin.URLs.WithCustomDomain(org.CustomDomain).ReportExport(org.Name, proj.Name, req.Report, ""),
-				EditUrl:   s.admin.URLs.WithCustomDomain(org.CustomDomain).ReportEdit(org.Name, proj.Name, req.Report, ""),
-			},
-		}, nil
-	}
+	urls := make(map[string]*adminv1.GetReportMetaResponse_URLs)
 
-	var externalEmails []string
-	for _, email := range req.EmailRecipients {
-		_, err := s.admin.DB.FindUserByEmail(ctx, email)
-		if err != nil {
-			if errors.Is(err, database.ErrNotFound) {
-				externalEmails = append(externalEmails, email)
-			} else {
-				return nil, fmt.Errorf("failed to find user by email: %w", err)
-			}
-		}
+	var recipients []string
+	recipients = append(recipients, req.EmailRecipients...)
+	if req.AnonRecipients {
+		// add empty email for slack and other notifiers token
+		recipients = append(recipients, "")
 	}
 
-	externalUrls := make(map[string]*adminv1.GetReportMetaResponse_URLs, len(externalEmails))
-	emailTokens, err := s.reconcileMagicTokensForExternalEmails(ctx, proj.OrganizationID, proj.ID, req.Report, req.OwnerId, externalEmails)
+	tokens, ownerEmail, err := s.createMagicTokens(ctx, proj.OrganizationID, proj.ID, req.Report, req.OwnerId, recipients, req.Resources)
```

Comment:
If a filter was applied to the report then that doesn't get propagated to the magic auth token here – so in theory, people can explore more data than that which the report filters by. Is that right? This is probably desirable in many cases, but not necessarily for external partners.

I wonder if we should perhaps add a flag to the report YAML and `CreateReport` API for whether or not to generate an open URL or only an export URL. E.g. it could be named `open_mode: none|full` (and then we could add support for a `filtered` mode later)